### PR TITLE
Fixes#9331: Added Power of Four Logarithm Easy Solution 

### LINKED ISSUE #9331: Want to Add Power of Four logic using Logarithm
### Feature description
Want to Add DSA Solution of this problem - Power of Four - using Logarithm

**Problem Statement - 
Given an integer n, return true if it is a power of four. Otherwise, return false.
An integer n is a power of four, if there exists an integer x such that n == 4x.**

Example 1:
Input: n = 16
Output: true

### PR DESCRIPTION
**Pull request by crocmons**
### Describe your change:
The intuition behind this code is based on the property of powers of four. 
In base-4 (quaternary) representation, any power of four will have only one '1' digit followed by zero or more '0' digits. 
For example, 4^0 = 1, 4^1 = 4, 4^2 = 16, and so on. 
In base-4, these numbers are represented as 1, 10, 100, and so on.
So, if you take the base-10 logarithm of a power of four and divide it by the base-10 logarithm of 4, you should get an integer. If you get an integer, it means the number is a power of four.

* [x] Add an algorithm?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #9331 ".

### CODE REVIEW

File: maths/power_of_four.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# Time Complexity:
+# The time complexity of this code is O(1) because it involves basic mathematical operations like logarithms and integer checks, which have constant time complexity.
+
+#  Space Complexity:
+#  The space complexity of this code is also O(1) because it uses a constant amount of memory to store the result of the logarithmic calculation and the integer check. The space used is not dependent on the input value n, so it remains constant regardless of the size of n.
+
+
+import math
+
+def isPowerOfFour(n):
```
[the remark this answers is not in the document]
Comment (crocmons) [reply maths/power_of_four.py:35]: resolved

File: maths/power_of_four.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# The time complexity of this code is O(1) because it involves basic mathematical operations like logarithms and integer checks, which have constant time complexity.
+
+#  Space Complexity:
+#  The space complexity of this code is also O(1) because it uses a constant amount of memory to store the result of the logarithmic calculation and the integer check. The space used is not dependent on the input value n, so it remains constant regardless of the size of n.
+
+
+import math
+
+
+def ispower_of_four(num):
```
[the remark this answers is not in the document]
Comment (crocmons) [reply maths/power_of_four.py:35]: added

**Comment by tianyizheng02** [comment]
See discussion in #9331

**Comment by crocmons** [comment]
> See discussion in #9331

yeap checked

**Pull request closed without merging**